### PR TITLE
Integrate reactions to messages in the new UI

### LINKED ISSUE #13523: UI Component - Reaction
This issue is to track the creation of the component: **Reaction**

You can access the link at https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Mobile-Components?node-id=3161%3A49379.

Please have a chat with the design team before starting with the component, and don't forget to add sizing to the issue

### PR DESCRIPTION
**Pull request by briansztamfater**
fixes #13523 

### Summary

Reaction component

#### Platforms

- Android
- iOS

#### Areas that maybe impacted

- Pinned messages probably

##### Functional

- 1-1 chats
- public chats
- group chats

### Steps to test

- Open Status
- Log in
- Open chat
- Add reactions and see other user reactions

status: wip

### CODE REVIEW

**Comment by ibrkhalil** [comment]
Hey @briansztamfater 
[This](https://github.com/status-im/status-mobile/pull/13631) PR might be helpful

**Comment by briansztamfater** [comment]
> Hey @briansztamfater [This](https://github.com/status-im/status-mobile/pull/13631) PR might be helpful

@ibrkhalil Definitely, I'll wait till that PR is merged then! Also wondering how both issues differ (#13523 and #13524), seem very similar cc @cammellos

**Comment by cammellos** [comment]
@briansztamfater good question, I created them through a script, not exactly sure if they are separate, we can probably keep one for now? (either is fine), sorry for the mixup

**Comment by briansztamfater** [comment]
@ibrkhalil do you think it is worth keeping this PR for integrating the component of your PR #13631 into messages in chat? If yes, I can wait till that PR is merged and then integrate it, if not I can close this

**Comment by du82** [comment]
it makes more sense to do it all in one rather than close this

**Comment by ibrkhalil** [comment]
> @ibrkhalil do you think it is worth keeping this PR for integrating the component of your PR #13631 into messages in chat? If yes, I can wait till that PR is merged and then integrate it, if not I can close this

I think keeping it is a good idea, we'd still need to do the integration anyway so why do it twice .. I'll merge mine today hopefully

**Comment by ibrkhalil** [comment]
Merged

**Comment by briansztamfater** [comment]
> Merged

🫡, I'll rebase and integrate it then

File: src/status_im/ui/screens/chat/message/reactions_row.cljs [hunk 1]
```diff
@@ -2,22 +2,17 @@
...
+(defn message-reactions [reactions timeline on-emoji-press on-open]
   (when (seq reactions)
-    [rn/view {:style (styles/reactions-row message timeline)}
-     (for [emoji-reaction reactions]
+    [rn/view {:style (styles/reactions-row timeline)}
+     (for [{:keys [own emoji-id quantity] :as emoji-reaction} reactions]
        ^{:key (str emoji-reaction)}
-       [reaction message emoji-reaction timeline])]))
+       [rn/view {:style {:margin-right 6 :margin-top 5}}
+        [quo2.react/render-react {:emoji (get constants/reactions emoji-id)
```
Comment (flexsurfer) [inline src/status_im/ui/screens/chat/message/reactions_row.cljs]: probably better to rename both ns and component, to not confuse with react, something like `quo2.reactions/reaction`

**Review by flexsurfer — APPROVED** [review]

**Comment by qoqobolo** [comment]
Hey @briansztamfater, thanks for PR!
Faced with an error during testing, can you check please. Can't reproduce this in develop. 

### ISSUE 1: Error when long-pressing on a message in new UI

**Steps:**
1. Enable new UI
2. Open any chat
3. Send a message
4. Long tap it to add a reaction

Screenshot 2022-09-14 at 16 22 27

**Comment by briansztamfater** [comment]
Hey @qoqobolo, thanks for testing. Some code had been refactored when we introduced the new message input for the new UI and I was able to reproduce the issue on my branch but also on develop (with iOS Simulator).

Said that, I made a quick fix for re-enabling reactions on the New UI, which may be a bit buggy since we have introduced some breaking changes, but won't be needed and will be replaced when we merge #13886.

Let me know how testing goes :)

File: src/status_im/ui/screens/chat/message/reactions_row.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/status_im/ui/screens/chat/message/reactions_row.cljs]: still not addressed

File: src/status_im/ui/screens/chat/message/reactions_row.cljs [hunk 2]
```diff
@@ -2,22 +2,17 @@
...
-   [quo/text {:accessibility-label (str "emoji-" emoji-id "-is-own-" own)
-              :style               (styles/reaction-quantity-style {:own own})}
-    quantity]])
-
-(defn message-reactions [message reactions timeline]
+(defn message-reactions [{:keys [content-type]} reactions timeline on-emoji-press on-open]
   (when (seq reactions)
-    [rn/view {:style (styles/reactions-row message timeline)}
-     (for [emoji-reaction reactions]
+    [rn/view {:style (styles/reactions-row timeline (if (= content-type constants/content-type-text) -3 5))}
```
Comment (flexsurfer) [inline src/status_im/ui/screens/chat/message/reactions_row.cljs]: magic

File: src/status_im/ui/screens/chat/message/styles.cljs [hunk 3]
```diff
@@ -84,6 +89,15 @@
            {:padding-left (+ 30 photos/default-size (when timeline 8))}
            {:padding-left 30})))
 
+(defn reactions-row [timeline margin-top]
```
Comment (flexsurfer) [inline src/status_im/ui/screens/chat/message/styles.cljs:92]: align map format

**Review by flexsurfer — DISMISSED** [review]

File: src/status_im/ui/screens/chat/message/reactions_row.cljs (hunk 1 above)
Comment (briansztamfater) [reply src/status_im/ui/screens/chat/message/reactions_row.cljs]: done

File: src/status_im/ui/screens/chat/message/message.cljs [hunk 4]
```diff
@@ -629,15 +629,13 @@
...
   [:<>
    [reactions/with-reaction-picker
     {:message         message
      :reactions       @(re-frame/subscribe [:chats/message-reactions (:message-id message) (:chat-id message)])
-     :picker-on-open  (fn []
-                        (space-keeper true))
-     :picker-on-close (fn []
-                        (space-keeper false))
+     :picker-on-open  (fn [])
+     :picker-on-close (fn [])
```
Comment (ibrkhalil) [inline src/status_im/ui/screens/chat/message/message.cljs:638]: Can be replace with `comment` which will always return `nil`
`:picker-on-open comment`

**Comment by qoqobolo** [comment]
Testing is blocked due to https://github.com/status-im/status-mobile/issues/14026 (the error occurs when creating a community with the new UI enabled _**or**_ after creating a community in the old UI and switching to the new one).
Let's wait until https://github.com/status-im/status-mobile/pull/14029 is merged.

**Comment by J-Son89** [comment]
> Testing is blocked due to #14026 (the error occurs when creating a community with the new UI enabled _**or**_ after creating a community in the old UI and switching to the new one). Let's wait until #14029 is merged.

will get that merged asap :)

**Comment by qoqobolo** [comment]
Thanks @J-Son89! No rush :)

**Comment by qoqobolo** [comment]
@briansztamfater as 14029 is merged, can you rebase this PR as well please?
Thanks

**Comment by briansztamfater** [comment]
Thanks @J-Son89 for merging and unblocking ;)

@qoqobolo rebase done :)

**Comment by qoqobolo** [comment]
Thanks @briansztamfater !

Hey @flexsurfer, could you clarify please if this PR is approved by you or still needs some review? I can see your approval but it was removed later by mistake if I understand correctly

**Review by flexsurfer — APPROVED** [review]

**Comment by pavloburykh** [comment]
@briansztamfater thank you for the PR. Tested, no regression found. Failed tests are not related to the PR. Ready for merge.

**Pull request merged**